src/jdk.compiler/share/classes/com/sun/tools/javac/comp/InferenceContext.java

```diff
@@ -552,7 +570,7 @@ List<Type> solveBasic(List<Type> varsToSolve, EnumSet<InferenceStep> steps) {
     @Override
     public String toString() {
         return "Inference vars: " + inferencevars + '\n' +
-               "Undet vars: " + undetvars;
+               "Undet vars: " + undetvars + '\n';
```

Comment:
Optional: should this also print the parent?

Reply:
sure, done